test/signalrclient-e2e-tests/hub_connection_tests.cpp

```diff
@@ -0,0 +1,184 @@
+// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#pragma once
+
+#include "stdafx.h"
+#include <string>
+#include "cpprest\details\basic_types.h"
+#include "cpprest\json.h" 
+#include "connection.h"
+#include "hub_connection.h"
+
+extern utility::string_t url;
+
+TEST(hub_connection_tests, connection_status_start_stop_start_reconnect)
+{
+    auto hub_conn = std::make_shared<signalr::hub_connection>(url + U("SignalR"));
+    auto weak_hub_conn = std::weak_ptr<signalr::hub_connection>(hub_conn);
+    auto reconnecting_event = std::make_shared<pplx::event>();
+    auto reconnected_event = std::make_shared<pplx::event>();
+
+    auto hub_proxy = hub_conn->create_hub_proxy(U("hubConnection"));
+
+    hub_conn->set_reconnecting([weak_hub_conn, reconnecting_event]()
+    {
+        auto conn = weak_hub_conn.lock();
+        if (conn)
+        {
+            ASSERT_EQ(conn->get_connection_state(), signalr::connection_state::reconnecting);
+        }
+        reconnecting_event->set();
+    });
+
+    hub_conn->set_reconnected([weak_hub_conn, reconnected_event]()
+    {
+        auto conn = weak_hub_conn.lock();
+        if (conn)
+        {
+            ASSERT_EQ(conn->get_connection_state(), signalr::connection_state::connected);
+        }
+        reconnected_event->set();
+    });
+
+    hub_conn->start().get();
+    ASSERT_EQ(hub_conn->get_connection_state(), signalr::connection_state::connected);
+
+    hub_conn->stop().get();
+    ASSERT_EQ(hub_conn->get_connection_state(), signalr::connection_state::disconnected);
+
+    hub_conn->start().get();
+    ASSERT_EQ(hub_conn->get_connection_state(), signalr::connection_state::connected);
+
+    try
```

Comment:
"forceReconnect" hub method throw to force reconnect, so here need to catch and test continue.